core/modules/admin_bar/admin_bar.inc

```diff
@@ -855,7 +855,7 @@ function admin_bar_theme_settings() {
     '#title' => t('Enabled components'),
     '#description' => t('These features will be enabled/visible in the administration bar. Untick the boxes to disable/hide them.'),
   );
-  $form['components']['components'] = array(
+  $form['components']['options'] = array(
     '#type' => 'checkboxes',
     '#title' => t('Enabled components'),
     '#title_display' => 'invisible',
```

Comment:
One way to solve the problem:

```suggestion
    '#title_display' => 'invisible',
    '#parents' => array('components'),
```

This makes it so that this element considers itself to not have any parents (other than itself), and saves to `$form_state['values']['components']`.

Alternatively, restore the form element key as `'components'`, even though it looks weird having double-nesting like that.

Reply:
@quicksketch simply adding `'#tree' => TRUE` to the fieldset fixes things. I've also fixed the `#states` for two of the options that had also regressed + added `#indentation` to them to make it obvious that they are dependent on their respective "parent" items.